ultravox/model/ultravox_model.py

```diff
@@ -162,8 +173,40 @@ def forward(
             past_key_values=past_key_values,
             **kwargs,
         )
-
-        return lm_output
+        if self.training:
+            if self.loss_config.loss_function == LossFunction.CrossEntropy:
+                return lm_output
+            elif self.loss_config.loss_function == LossFunction.KL_Divergence:
+                # compute the teacher (text-only) model's distribution
```

Comment:
suggest hoisting this into a `self._compute_kl_loss()` function to try to keep the core `forward()` logic clear

Reply:
Makes sense. Will update